File: Day31/Java/string_Palindrome_recursion.java
public class string_Palindrome_recursion {
    static boolean checkpalindrome(int i,String str){
        if(i>=(str.length()/2)) return true;
        else if(str.charAt(i)!=str.charAt(str.length()-i-1)) return false;
        return checkpalindrome(i+1,str);
    }
    static boolean isPalindrome(String str) {
        return checkpalindrome(0,str);
    }
    public static void main(String[] args){
        String str = "madam";
        boolean ans = (isPalindrome(str));
        System.out.println(ans);
    }
    
}

File: Day31/CPP/string_Palindrome_recursion.cpp
#include<bits/stdc++.h>
using namespace std;
bool checkpalindrome(int i,string &str,int n){
    if(i>=(n/2)) return true;
    else if(str[i]!=str[n-i-1]) return false;
    checkpalindrome(i+1,str,n);
}
bool isPalindrome(string& str) {
    return checkpalindrome(0,str,str.length());
}

int main()
{
    string str = "madam";
    bool ans =  isPalindrome(str);
    cout<<ans;
    return 0;
}